! Multi Contas / Mapeamento de Baus / Escolha Heroes / Correções

bnudaniel22 opened the pull request:
-Funcionalidades
Multi Contas

Melhorias na seleção de heróis (Mapeamento de Baus / Heroes por raridade)
Melhor apresentação informação de Baús
Parâmetro escolher_baus_heroes: True para ativa escolha de baus e heroes. / False = funciona como versão original
Melhoria na contagem de Baús e seleção de heróis.
Parâmetro mandar_todos_trabalhar = True ele não analisa os Baus, apenas clica no ALL sempre
Parâmetro para Maximizar janelas
Ex.bat - execução em looping.
Log em inglês

-Se (if)
Full Stamina - Manda todos trabalhar
+9 descansando faz seleção novamente

-Correção
Scroll - Atualização Imagens(tamanho)
Click Ok - quando desconectado
Correção ao clicar no botão Work

-Outros

Junção de codigos
AnderRobson:fea-duasconta / AnderRobson:fea-runningforever / AnderRobson:fea-heroesrotatio

bnudaniel22 committed:
Multi Contas + Correções

-Funcionalidades
  Multi Contas
  Escolha (Mapeamento de Baus / Heroes por raridade)
  Ex.bat - execução em looping.

-Se (if)
  Full Stamina - Manda todos trabalhar
  +9 descansando faz seleção novamente

-Correção
  Scroll - Atualização Imagens(tamanho)
  Click Ok - quando desconectado

-Outros

Junção de codigos
AnderRobson:fea-duasconta / AnderRobson:fea-runningforever / AnderRobson:fea-heroesrotation

bnudaniel22 committed:
Correção Seleção heroes

bnudaniel22 committed:
Revert "Multi Contas + Correções"

This reverts commit fcf464324a359c91b6eaad2271983c9db5f30396.

bnudaniel22 committed:
Revert "Correção Seleção heroes"

This reverts commit 348e2afbb98f9529d9637634a7ce2d5a7e4153de.

bnudaniel22 committed:
Multi Contas / Mapeamento de Baus / Escolha Heroes / Correções

-Funcionalidades
Multi Contas
Escolha (Mapeamento de Baus / Heroes por raridade)
Ex.bat - execução em looping.

-Se (if)
Full Stamina - Manda todos trabalhar
+9 descansando faz seleção novamente

-Correção
Scroll - Atualização Imagens(tamanho)
Click Ok - quando desconectado

-Outros

Junção de codigos
AnderRobson:fea-duasconta / AnderRobson:fea-runningforever / AnderRobson:fea-heroesrotatio

bnudaniel22 committed:
Revert "Multi Contas / Mapeamento de Baus / Escolha Heroes / Correções"

This reverts commit cf71110d96987d2606ade3aabb81228f07d33a4f.

bnudaniel22 committed:
! Multi Contas / Mapeamento de Baus / Escolha Heroes / Correções

-Funcionalidades
Multi Contas
Escolha (Mapeamento de Baus / Heroes por raridade)
Ex.bat - execução em looping.

-Se (if)
Full Stamina - Manda todos trabalhar
+9 descansando faz seleção novamente

-Correção
Scroll - Atualização Imagens(tamanho)
Click Ok - quando desconectado

-Outros

Junção de codigos
AnderRobson:fea-duasconta / AnderRobson:fea-runningforever / AnderRobson:fea-heroesrotatio

kendikaku commented:
nao funciona, da erro na hora de executar o bot

bnudaniel22 commented:
> nao funciona, da erro na hora de executar o bot

Que tipo de erro?  Como o bot fica transitando entre as multi contas(Navegadores) o computador tem de ser exclusivo pros jogos. Não pode usar durante o jogo. 
Para executar abra o arquivo ex.bat (para chamar o programa em looping).

kendikaku commented:
> > nao funciona, da erro na hora de executar o bot
> 
> Que tipo de erro? Como o bot fica transitando entre as multi contas(Navegadores) o computador tem de ser exclusivo pros jogos. Não pode usar durante o jogo. Para executar abra o arquivo ex.bat (para chamar o programa em looping).

Traceback (most recent call last):
  File "C:\Users\richa\Desktop\bombcrypto-bot-main\bombcrypto-bot-main\index.py", line 719, in <module>
    main()
  File "C:\Users\richa\Desktop\bombcrypto-bot-main\bombcrypto-bot-main\index.py", line 624, in main
    last["window"].activate()
  File "C:\Users\richa\AppData\Local\Programs\Python\Python310\lib\site-packages\pygetwindow\_pygetwindow_win.py", line 246, in activate
    _raiseWithLastError()
  File "C:\Users\richa\AppData\Local\Programs\Python\Python310\lib\site-packages\pygetwindow\_pygetwindow_win.py", line 99, in _raiseWithLastError
    raise PyGetWindowException('Error code from Windows: %s - %s' % (errorCode, _formatMessage(errorCode)))
pygetwindow.PyGetWindowException: Error code from Windows: 0 - A operação foi concluída com êxito.

bnudaniel22 commented:
> > > nao funciona, da erro na hora de executar o bot
> > 
> > 
> > Que tipo de erro? Como o bot fica transitando entre as multi contas(Navegadores) o computador tem de ser exclusivo pros jogos. Não pode usar durante o jogo. Para executar abra o arquivo ex.bat (para chamar o programa em looping).
> 
> Traceback (most recent call last): File "C:\Users\richa\Desktop\bombcrypto-bot-main\bombcrypto-bot-main\index.py", line 719, in main() File "C:\Users\richa\Desktop\bombcrypto-bot-main\bombcrypto-bot-main\index.py", line 624, in main last["window"].activate() File "C:\Users\richa\AppData\Local\Programs\Python\Python310\lib\site-packages\pygetwindow_pygetwindow_win.py", line 246, in activate _raiseWithLastError() File "C:\Users\richa\AppData\Local\Programs\Python\Python310\lib\site-packages\pygetwindow_pygetwindow_win.py", line 99, in _raiseWithLastError raise PyGetWindowException('Error code from Windows: %s - %s' % (errorCode, _formatMessage(errorCode))) pygetwindow.PyGetWindowException: Error code from Windows: 0 - A operação foi concluída com êxito.

Como falei anteriormente, chama o bot pelo ex.bat (não pelo index.py). O Ex.bat chama o index.py em looping mesmo ocorrendo essas exceções. 
As exceções ocorrem quando o computador é utilizado enquanto o bot esta fazendo as jogadas. Mesmo dando esses erros o looping chama novamente bot e começa novamente a rotina.
Utilizo com 4 contas e esta rodando de boa.

bnudaniel22 committed:
Correções

Quando enviar heroes para trabalhar novamente.

kendikaku commented:
> > > > nao funciona, da erro na hora de executar o bot
> > > 
> > > 
> > > Que tipo de erro? Como o bot fica transitando entre as multi contas(Navegadores) o computador tem de ser exclusivo pros jogos. Não pode usar durante o jogo. Para executar abra o arquivo ex.bat (para chamar o programa em looping).
> > 
> > 
> > Traceback (most recent call last): File "C:\Users\richa\Desktop\bombcrypto-bot-main\bombcrypto-bot-main\index.py", line 719, in main() File "C:\Users\richa\Desktop\bombcrypto-bot-main\bombcrypto-bot-main\index.py", line 624, in main last["window"].activate() File "C:\Users\richa\AppData\Local\Programs\Python\Python310\lib\site-packages\pygetwindow_pygetwindow_win.py", line 246, in activate _raiseWithLastError() File "C:\Users\richa\AppData\Local\Programs\Python\Python310\lib\site-packages\pygetwindow_pygetwindow_win.py", line 99, in _raiseWithLastError raise PyGetWindowException('Error code from Windows: %s - %s' % (errorCode, _formatMessage(errorCode))) pygetwindow.PyGetWindowException: Error code from Windows: 0 - A operação foi concluída com êxito.
> 
> Como falei anteriormente, chama o bot pelo ex.bat (não pelo index.py). O Ex.bat chama o index.py em looping mesmo ocorrendo essas exceções. As exceções ocorrem quando o computador é utilizado enquanto o bot esta fazendo as jogadas. Mesmo dando esses erros o looping chama novamente bot e começa novamente a rotina. Utilizo com 4 contas e esta rodando de boa.

Não adianta, pelo ex.bat, fica em um loop infinito de erro é depois erro novamente.

KissCallMe commented:
Desculpe ser tão leigo, mas onde baixo essa sua versão modificada?

kendikaku commented:
> Desculpe ser tão leigo, mas onde baixo essa sua versão modificada?

https://github.com/bnudaniel22/bombcrypto-bot/archive/refs/heads/main.zip

KissCallMe commented:
Não tem como configurar o BOT para quando escolher a opção para mandar todos os bonecos (all) ele simplesmente clickar naquele botão novo?

bnudaniel22 commented:
> Não tem como configurar o BOT para quando escolher a opção para mandar todos os bonecos (all) ele simplesmente clickar naquele botão novo?

Atualizado bot
Adicionado parâmetro no config.yaml - coloca assim = mandar_todos_trabalhar: True

bnudaniel22 committed:
+Add Funcionalidade

Mandar todos trabalha sempre
 -Marcar mandar_todos_trabalhar: True no config.yaml

mopped7 commented:
Opa, boa tarde! Desde já agradeço por esse update...
Porém tive dúvidas em duas coisas adicionadas...

Escolha (Mapeamento de Baus / Heroes por raridade)
Ex.bat - execução em looping.

Como funcionaria esse mapeamento de baús/herois por raridade?
Ele entenderia quantos baús possuem e colocaria os herois certos para trabalhar?
E pq a questão do looping? o código normal já não estaria em loop? abraços irmão, obrigado!

ghost commented:
Daniel, obrigado pela melhoria da ferramenta.

Tentei executar na minha maquina virtual linux e não funciona. Gostaria de saber se estou fazendo algo de errado ou se realmente não funciona em Linux.

![Screenshot_5](https://user-images.githubusercontent.com/39542354/149676563-ed66f2c0-96ca-451b-8d9b-25ca3d39ab62.png)

bnudaniel22 commented:
> Daniel, obrigado pela melhoria da ferramenta.
> 
> Tentei executar na minha maquina virtual linux e não funciona. Gostaria de saber se estou fazendo algo de errado ou se realmente não funciona em Linux.
> 
> ![Screenshot_5](https://user-images.githubusercontent.com/39542354/149676563-ed66f2c0-96ca-451b-8d9b-25ca3d39ab62.png)

Realmente não é pra funcionar em linux, pq tem diferenças entre as janelas dos sistemas pelo que vi no código.
Tem um post nesse link que fala sobre linux. https://github.com/samuelndm/bombcrypto-bot. Deve funcionar. 
Talvez juntando os códigos venha a funcionar pra vc.

bnudaniel22 commented:
> Opa, boa tarde! Desde já agradeço por esse update... Porém tive dúvidas em duas coisas adicionadas...
> 
> Escolha (Mapeamento de Baus / Heroes por raridade) Ex.bat - execução em looping.
> 
> Como funcionaria esse mapeamento de baús/herois por raridade? Ele entenderia quantos baús possuem e colocaria os herois certos para trabalhar? E pq a questão do looping? o código normal já não estaria em loop? abraços irmão, obrigado!

Sim ele lê os Baus e decide quais herois vão trabalhar, é um código em teste do "AnderRobson", link dele https://github.com/mpcabete/bombcrypto-bot/pull/512

Referente ao looping (ocorrem exceções "last["window"].activate()") - O Ex.bat chama o index.py em looping mesmo ocorrendo essas exceções. 
As exceções ocorrem quando o computador é utilizado enquanto o bot esta fazendo as jogadas. Mesmo dando esses erros o looping chama novamente bot e começa novamente a rotina.
Quando o computador é exclusivo pro jogo, e o bot consegue transitar entre os navegadores sem interferência do usuário ai as exceções não acontecem.

moreirasthiago commented:
Opa boa noite,

Esta versão ela funciona praticamente como a "principal"? 

Tipo verificando os Heroes que estão com a barra verde e colocando pra trabalhar ?

jeffersonmdias commented:
Por algum motivo ele ta ignorando alguns herois com barra verde, e queria tirar uma dúvida sobre os baús como faz para utilizar essa função  e como ela funciona realmente, ele analisa os baus e manda herois fortes ou fracos dependendo do hp do bau?
Preciso ativar alguma parte no codigo para essa função do bau funcionar?

ghost commented:
Não sou programador mas tenho uma pequena noção e Tentei fazer a portabilidade de parte do código do @bnudaniel22 para o oficial @mpcabete (que é o que funciona pra mim) e não consegui... A versão do windows não funciona na minha maquina apesar rodar o programa não executa os clicks na tela. Achei muito legal a função que analisa os báus, porém, como mencionei anteriomente, não consegui fazer rodar :(

Alguma alma bondosa poderia fazer essa portabilidade? @bnudaniel22 vc tem discord?

bnudaniel22 commented:
> Por algum motivo ele ta ignorando alguns herois com barra verde, e queria tirar uma dúvida sobre os baús como faz para utilizar essa função e como ela funciona realmente, ele analisa os baus e manda herois fortes ou fracos dependendo do hp do bau? Preciso ativar alguma parte no codigo para essa função do bau funcionar?

Conforme falei é um código teste do "AnderRobson"
Ele trata os baus nas seguintes funções (Faz um calculo pra decidir mandar ou não os Heróis)

def checkBaus():
  .....
    if (len(bausRoxoMeio) > 0 or len(bausGoldMeio) > 0 or len(bausBlueMeio) > 0):
        response = (
                           (len(bausRoxoMeio) + len(bausGoldMeio) + len(bausBlueMeio))
                           * 100
                   ) / (len(bausRoxo) + len(bausGold) + len(bausBlue))
    else:
        response = 0



def clickGreenBarButtons(baus):

        deveTrabalhar = 1

        if raridade != 'commom' and (baus >= 60 or baus == 0):
            deveTrabalhar = 0


Pelo que entendi ele manda as Raridades quando os Baus estão mais fortes

bnudaniel22 commented:
> Opa boa noite,
> 
> Esta versão ela funciona praticamente como a "principal"?
> 
> Tipo verificando os Heroes que estão com a barra verde e colocando pra trabalhar ?

Ele faz uma analise dos Baus do mapa e manda determinados heróis trabalhar
Conforme falei é um código teste do "AnderRobson"
Ele trata os baus nas seguintes funções (Faz um calculo pra decidir mandar ou não os Heróis)

def checkBaus():
  .....
    if (len(bausRoxoMeio) > 0 or len(bausGoldMeio) > 0 or len(bausBlueMeio) > 0):
        response = (
                           (len(bausRoxoMeio) + len(bausGoldMeio) + len(bausBlueMeio))
                           * 100
                   ) / (len(bausRoxo) + len(bausGold) + len(bausBlue))
    else:
        response = 0



def clickGreenBarButtons(baus):
  .....
        deveTrabalhar = 1

        if raridade != 'commom' and (baus >= 60 or baus == 0):
            deveTrabalhar = 0

bnudaniel22 commented:
> Não sou programador mas tenho uma pequena noção e Tentei fazer a portabilidade de parte do código do @bnudaniel22 para o oficial @mpcabete (que é o que funciona pra mim) e não consegui... A versão do windows não funciona na minha maquina apesar rodar o programa não executa os clicks na tela. Achei muito legal a função que analisa os báus, porém, como mencionei anteriomente, não consegui fazer rodar :(
> 
> Alguma alma bondosa poderia fazer essa portabilidade? @bnudaniel22 vc tem discord?

Ola qual versão do bot vc utiliza no linux? Me manda o link pra baixar e vejo quais as diferenças, soh não posso testar pois não estou usando linux no momento.

ghost commented:
> > Não sou programador mas tenho uma pequena noção e Tentei fazer a portabilidade de parte do código do @bnudaniel22 para o oficial @mpcabete (que é o que funciona pra mim) e não consegui... A versão do windows não funciona na minha maquina apesar rodar o programa não executa os clicks na tela. Achei muito legal a função que analisa os báus, porém, como mencionei anteriomente, não consegui fazer rodar :(
> > Alguma alma bondosa poderia fazer essa portabilidade? @bnudaniel22 vc tem discord?
> 
> Ola qual versão do bot vc utiliza no linux? Me manda o link pra baixar e vejo quais as diferenças, soh não posso testar pois não estou usando linux no momento.

Daniel, bom dia. 

Uso essa https://github.com/mpcabete/bombcrypto-bot/archive/refs/heads/main.zip - e funciona perfeitamente no linux, se conseguisse portar as funções de checkBaus seria top!

bnudaniel22 commented:
> > > Não sou programador mas tenho uma pequena noção e Tentei fazer a portabilidade de parte do código do @bnudaniel22 para o oficial @mpcabete (que é o que funciona pra mim) e não consegui... A versão do windows não funciona na minha maquina apesar rodar o programa não executa os clicks na tela. Achei muito legal a função que analisa os báus, porém, como mencionei anteriomente, não consegui fazer rodar :(
> > > Alguma alma bondosa poderia fazer essa portabilidade? @bnudaniel22 vc tem discord?
> > 
> > 
> > Ola qual versão do bot vc utiliza no linux? Me manda o link pra baixar e vejo quais as diferenças, soh não posso testar pois não estou usando linux no momento.
> 
> Daniel, bom dia.
> 
> Uso essa https://github.com/mpcabete/bombcrypto-bot/archive/refs/heads/main.zip - e funciona perfeitamente no linux, se conseguisse portar as funções de checkBaus seria top!

Verifica se funciona, segue link https://www.dropbox.com/s/i9kvl49sku12zcj/BombBot.zip?dl=0

ghost commented:
> > > > Não sou programador mas tenho uma pequena noção e Tentei fazer a portabilidade de parte do código do @bnudaniel22 para o oficial @mpcabete (que é o que funciona pra mim) e não consegui... A versão do windows não funciona na minha maquina apesar rodar o programa não executa os clicks na tela. Achei muito legal a função que analisa os báus, porém, como mencionei anteriomente, não consegui fazer rodar :(
> > > > Alguma alma bondosa poderia fazer essa portabilidade? @bnudaniel22 vc tem discord?
> > > 
> > > 
> > > Ola qual versão do bot vc utiliza no linux? Me manda o link pra baixar e vejo quais as diferenças, soh não posso testar pois não estou usando linux no momento.
> > 
> > 
> > Daniel, bom dia.
> > Uso essa https://github.com/mpcabete/bombcrypto-bot/archive/refs/heads/main.zip - e funciona perfeitamente no linux, se conseguisse portar as funções de checkBaus seria top!
> 
> Verifica se funciona, segue link https://www.dropbox.com/s/i9kvl49sku12zcj/BombBot.zip?dl=0

Daniel, muito obrigado! Segue o feedback

O Bot funcionou, porém, ao iniciar ele está triggando ao mesmo tempo a função "search for heroes to work" e "search for baus to map" ao mesmo tempo. Isso poderia talvez esteja causando o erro na leitura dos baus?

Acho que não está tendo habil para o bot ler todas as informações de baus no mapa, pois, ele lê apenas alguns baus roxo (e valor incorreto, pois há mais no mapa). Os baus gold e o azul ele não está lendo. Seria possivel colocar um delay para na hora que trigasse o checkBaus ele demorasse um pouco mais para trazer a quantidade/tipo de baus correta?

Houve um problema com a leitura do hp dos herois (hp verde), a imagem green-bar.png do seu arquivo não funcionou na leitura e não disparou o click para eles trabalhassem. Coloquei a imagem green-bar.png do projeto original e consegui corrigir isso! (y)

Como a leitura dos baús não está funcionando corretamente o bot **não está** colocando meu Épico e o Super raro para trabalhar, mesmo estando com hp full/green. Isso pode está ocorrendo porque ele não está lendo a informações de bau corretas? O projeto original do @mpcabete faz essa leitura da escolha dos herois de forma mais lenta e funfa 100%

A Leitura dos herois que estão dormindo também está incorreta.

Estou anexando o log do bot funcionando por volta de 15 minutos para melhor entendimento e Screenshot da tela.

Desde já agradeço pela colaboração

OBS: Uso VPS da oracle, talvez tenha um delay na comunicação do servidor do bomb.
![Screenshot_1](https://user-images.githubusercontent.com/39542354/149806820-826169a2-7f54-494f-8801-9dadb0388403.png)
[logger.log](https://github.com/mpcabete/bombcrypto-bot/files/7883405/logger.log)

!

raphaaaodev commented:
Para ocorrer uma melhor verificação dos baús, é necessário que o bot coloque todos os heróis para trabalhar, evitando que o balão que representa o descanso fique na frente de algum baú.

Por não ser fixo o balão de descanso e a imagem dos baús (varia a vida). Isso pode acabar afetando da leitura das screenshots.

Talvez seja necessário alterar as imagens ou trabalhar com o valor do threshold.

ghost commented:
> Para ocorrer uma melhor verificação dos baús, é necessário que o bot coloque todos os heróis para trabalhar, evitando que o balão que representa o descanso fique na frente de algum baú.
> 
> Por não ser fixo o balão de descanso e a imagem dos baús (varia a vida). Isso pode acabar afetando da leitura das screenshots.
> 
> Talvez seja necessário alterar as imagens ou trabalhar com o valor do threshold.

Faz sentido, quanto a imagem mais clean melhor na leitura, porém, quando "mandar_todos_trabalhar = true" no config.yaml o bot não ativa a função checar baus

```
if c['mandar_todos_trabalhar'] == False:
        baus = checkBaus()
```
Vou tentar mexer no threshold... Último caso mexo nas imagens dos baús

bnudaniel22 commented:
> Ele faz uma analise dos Baus do mapa e manda determinados heróis trabalhar
> Conforme falei é um código teste do "AnderRobson"

Ele faz uma analise dos Baus do mapa e manda determinados heróis trabalhar
Conforme falei é um código teste do "AnderRobson"
Quanto a precisão da detecção das imagens (Baus e Dormindo) não vou saber responder (Rotina não foi criada por mim).

Ele trata os baus nas seguintes funções (Faz um calculo pra decidir mandar ou não os Heróis) 
Por esse motivo as raridades algumas vezes não são enviadas pro trabalho (Percebi isto e mudei pra quando achar um Heroe full stamina, ele clica no All e manda todos trabalhar)

def checkBaus():
.....
if (len(bausRoxoMeio) > 0 or len(bausGoldMeio) > 0 or len(bausBlueMeio) > 0):
response = (
(len(bausRoxoMeio) + len(bausGoldMeio) + len(bausBlueMeio))
* 100
) / (len(bausRoxo) + len(bausGold) + len(bausBlue))
else:
response = 0

def clickGreenBarButtons(baus):
.....
    deveTrabalhar = 1

    if raridade != 'commom' and (baus >= 60 or baus == 0):
        deveTrabalhar = 0


Pelo que entendi ele manda as Raridades quando os Baus estão mais fortes

bnudaniel22 commented:
> mandar_todos_trabalhar = true

mandar_todos_trabalhar = True ele não analisa os Baus, apenas clica no ALL sempre

YanHappy commented:
Can you change the bot work without maximizing the window when working ?. I realize that if we maximize all windows, the hero will get stuck sometimes and doesn't work as expected. When it compares the income, your version works smooth but income reduces from 20bc / day to 15-16bc / day.

bnudaniel22 committed:
-Funcionalidades / Ativar|Desativar Escolha Baus

Parametro escolher_baus_heroes: True para ativa escolha de baus e heroes.
False = funciona como versão original

ghost commented:
Muito obrigado, @bnudaniel22 e @raphaaaodev 

Fiz alteração (diminui o valor) na threshold e melhorou no reconhecimento dos baus.

kendikaku commented:
> Muito obrigado, @bnudaniel22 e @raphaaaodev
> 
> Fiz alteração (diminui o valor) na threshold e melhorou no reconhecimento dos baus.

Compartilhe com a gente as suas config,

ghost commented:
> > Muito obrigado, @bnudaniel22 e @raphaaaodev
> > Fiz alteração (diminui o valor) na threshold e melhorou no reconhecimento dos baus.
> 
> Compartilhe com a gente as suas config,
[config.zip](https://github.com/mpcabete/bombcrypto-bot/files/7892264/config.zip)

@Japa18244, essa config é que atualmente uso. Pode ser que não funcione corretamente no seu PC, testa aí!

@bnudaniel22 Valeu demais pela ajuda!! Tmj

falpz commented:
Qaul o maximo de acc que suporta esse bot? estou com duas ta rodando legal porem to abrindo uma terceira conta.

ArturllVale commented:
Olá amigo, executo o Bot, e na hora de selecionar o personagem ele não faz nem uma ação seguinte... aparece somente : "Wait select heroes to work Manual."

Dassuen commented:
![image](https://user-images.githubusercontent.com/45463201/150092144-dcb1bc53-7bfa-4e3c-8f31-9e2d72964fa9.png)

Assim que ele seleciona para colocar os heroes para trabalhar ele da erro e reinicia o bot, ficando em um loop infinito, tentei as config do rapaz ai de cima mas nao adiantou, bot pra mim nao esta funcionando.

bnudaniel22 committed:
+Funcionalidades / Correções

-Parâmetro para Maximizar janelas
-Melhoria na contagem de Baús e seleção de heróis.

bnudaniel22 commented:
> Qaul o maximo de acc que suporta esse bot? estou com duas ta rodando legal porem to abrindo uma terceira conta.

To usando com 4 contas

moreirasthiago commented:
@bnudaniel22  ele funciona como a versão main do bot?  tipo verificando os que estão com barra verde e colocando pra trabalhar ?

bnudaniel22 committed:
Correção

bnudaniel22 commented:
> ![image](https://user-images.githubusercontent.com/45463201/150092144-dcb1bc53-7bfa-4e3c-8f31-9e2d72964fa9.png)
> 
> Assim que ele seleciona para colocar os heroes para trabalhar ele da erro e reinicia o bot, ficando em um loop infinito, tentei as config do rapaz ai de cima mas nao adiantou, bot pra mim nao esta funcionando.

Obrigado, realmente tinha um problema na questão apontada. 
O bot funciona automaticamente, qualquer interação do usuário ocasiona exceções que fazem o Bot reiniciar. Conecta na conta de deixe no automático. Tente novamente.

bnudaniel22 commented:
> -Funcionalidades Multi Contas Escolha (Mapeamento de Baus / Heroes por raridade) Parâmetro escolher_baus_heroes: True para ativa escolha de baus e heroes. / False = funciona como versão original Melhoria na contagem de Baús e seleção de heróis. Parâmetro mandar_todos_trabalhar = True ele não analisa os Baus, apenas clica no ALL sempre Parâmetro para Maximizar janelas Ex.bat - execução em looping.
> 
> -Se (if) Full Stamina - Manda todos trabalhar +9 descansando faz seleção novamente
> 
> -Correção Scroll - Atualização Imagens(tamanho) Click Ok - quando desconectado

É baseado na versão Main, porem com modificações e junção de outras versões.

-Funcionalidades
Multi Contas
Escolha (Mapeamento de Baus / Heroes por raridade)
Parâmetro escolher_baus_heroes: True para ativa escolha de baus e heroes. / False = funciona como versão original
Melhoria na contagem de Baús e seleção de heróis.
Parâmetro mandar_todos_trabalhar = True ele não analisa os Baus, apenas clica no ALL sempre
Parâmetro para Maximizar janelas
Ex.bat - execução em looping.

-Se (if)
Full Stamina - Manda todos trabalhar
+9 descansando faz seleção novamente

-Correção
Scroll - Atualização Imagens(tamanho)
Click Ok - quando desconectado

bnudaniel22 commented:
> Olá amigo, executo o Bot, e na hora de selecionar o personagem ele não faz nem uma ação seguinte... aparece somente : "Wait select heroes to work Manual."

Obrigado, realmente tinha um problema na questão apontada.
O bot funciona automaticamente, qualquer interação do usuário ocasiona exceções que fazem o Bot reiniciar. Conecta na conta de deixe no automático. Tente novamente.

moreirasthiago commented:
Pra executar eu tenho que deixar as janelas do navegador na tela ou dar start e ir dando click em cada uma delas ?

bnudaniel22 commented:
> ![image](https://user-images.githubusercontent.com/45463201/150092144-dcb1bc53-7bfa-4e3c-8f31-9e2d72964fa9.png)
> 
> Assim que ele seleciona para colocar os heroes para trabalhar ele da erro e reinicia o bot, ficando em um loop infinito, tentei as config do rapaz ai de cima mas nao adiantou, bot pra mim nao esta funcionando.

Estranho, aparece alguma exceção ou só isso q enviou na imagem? fiz algumas atualizações no código, tente novamente.

bnudaniel22 commented:
> Ex.bat

execua

bnudaniel22 commented:
> Pra executar eu tenho que deixar as janelas do navegador na tela ou dar start e ir dando click em cada uma delas ?

Executo o Ex.bat e clico em todos os navegadores com o jogo (1 por 1), depois de clicar em todos  o bot vai no automático. 
Depois não pode mais mexer no computador.  
Qualquer interação do usuário ocasiona exceções que faz o Bot reiniciar.

kendikaku commented:
Teve mais alguma melhoria a respeito dos baú e do envio específico dos herois?

bnudaniel22 commented:
> Teve mais alguma melhoria a respeito dos baú e do envio específico dos herois?

+Funcionalidades / Correções
-Parâmetro para Maximizar janelas
-Melhoria na contagem de Baús e seleção de heróis.

xReaperGG commented:
File "C:\Users\MBMR\3D Objects\bombcrypto-bot-main\index.py", line 55
    """Returns n with randomness
       ^
SyntaxError: invalid syntax

xReaperGG commented:
File "C:\Users\MBMR\3D Objects\bombcrypto-bot-main\index.py", line 55
    """Returns n with randomness
       ^
SyntaxError: invalid syntax

Sabe corrigir esse erro? Com a versao original do bot roda normalmente, mas com essa sempre me retorna esse erro  quando inicio.
Obs: Estou tentando iniciar pelo .bat

bnudaniel22 commented:
> index.py

Você esta usando o arquivo baixado ou juntou com outro?
Abra no notepad o index.py e verifique se esta assim.

def addRandomness(n, randomn_factor_size=None):
    """Returns n with randomness
    Parameters:
        n (int): A decimal integer
        randomn_factor_size (int): The maximum value+- of randomness that will be
            added to n

    Returns:
        int: n with randomness
    """

    if randomn_factor_size is None:
        randomness_percentage = 0.1
        randomn_factor_size = randomness_percentage * n

    random_factor = 2 * random() * randomn_factor_size
    if random_factor > 5:
        random_factor = 5
    without_average_random_factor = n - randomn_factor_size
    randomized_n = int(without_average_random_factor + random_factor)
    # logger('{} with randomness -> {}'.format(int(n), randomized_n))
    return int(randomized_n)

bnudaniel22 commented:
> File "C:\Users\MBMR\3D Objects\bombcrypto-bot-main\index.py", line 55 """Returns n with randomness ^ SyntaxError: invalid syntax
> 
> Sabe corrigir esse erro? Com a versao original do bot roda normalmente, mas com essa sempre me retorna esse erro quando inicio. Obs: Estou tentando iniciar pelo .bat

Você esta usando o arquivo baixado ou juntou com outro?
Abra no notepad o index.py e verifique se esta assim.

def addRandomness(n, randomn_factor_size=None):
"""Returns n with randomness
Parameters:
n (int): A decimal integer
randomn_factor_size (int): The maximum value+- of randomness that will be
added to n

xReaperGG commented:
> > File "C:\Users\MBMR\3D Objects\bombcrypto-bot-main\index.py", line 55 """Returns n with randomness ^ SyntaxError: invalid syntax
> > Sabe corrigir esse erro? Com a versao original do bot roda normalmente, mas com essa sempre me retorna esse erro quando inicio. Obs: Estou tentando iniciar pelo .bat
> 
> Você esta usando o arquivo baixado ou juntou com outro? Abra no notepad o index.py e verifique se esta assim.
> 
> def addRandomness(n, randomn_factor_size=None): """Returns n with randomness Parameters: n (int): A decimal integer randomn_factor_size (int): The maximum value+- of randomness that will be added to n

Baixei o arquivo direto mesmo, e o código está idêntico a esse, mas o erro continua.

TharkeD commented:
Boa noite, Primeiro Parabens pelo codigo tanto pra vce quanto pro anderson.
Porem tenho uma duvida se esse codigo é possivel selecionar o raridade do heroi pra mandar farmar por hora,
por exemplo, enviar commons a cada 40min raro =1hr sr =2hr legend =8hrs . e se ja tiver algum  codigo desse por ai ja peço desculpas pelo inconveniente
Abrç

bnudaniel22 commented:
> > > File "C:\Users\MBMR\3D Objects\bombcrypto-bot-main\index.py", line 55 """Returns n with randomness ^ SyntaxError: invalid syntax
> > > Sabe corrigir esse erro? Com a versao original do bot roda normalmente, mas com essa sempre me retorna esse erro quando inicio. Obs: Estou tentando iniciar pelo .bat
> > 
> > 
> > Você esta usando o arquivo baixado ou juntou com outro? Abra no notepad o index.py e verifique se esta assim.
> > def addRandomness(n, randomn_factor_size=None): """Returns n with randomness Parameters: n (int): A decimal integer randomn_factor_size (int): The maximum value+- of randomness that will be added to n
> 
> Baixei o arquivo direto mesmo, e o código está idêntico a esse, mas o erro continua.

Resolvido, erro meu. Foi mal

bnudaniel22 commented:
> Boa noite, Primeiro Parabens pelo codigo tanto pra vce quanto pro anderson. Porem tenho uma duvida se esse codigo é possivel selecionar o raridade do heroi pra mandar farmar por hora, por exemplo, enviar commons a cada 40min raro =1hr sr =2hr legend =8hrs . e se ja tiver algum codigo desse por ai ja peço desculpas pelo inconveniente Abrç

Ola, ainda não tem nada nesse sentido.

ipucoinadmin commented:
Ele funciona com um pc com mais de uma tela ?
@edit
testei e não funcionou

bnudaniel22 commented:
Conforme já mencionado acima o computador tem de ficar exclusivo pro bot durante o jogo multiconta. Utilizando 2 telas e usando o computador ao mesmo tempo o bot deve apresentar exceções reiniciado a rotina do bot.

Obter o Outlook para Android<https://aka.ms/AAb9ysg>
________________________________
From: ipucoinadmin ***@***.***>
Sent: Friday, January 21, 2022 12:03:39 PM
To: mpcabete/bombcrypto-bot ***@***.***>
Cc: bnudaniel22 ***@***.***>; State change ***@***.***>
Subject: Re: [mpcabete/bombcrypto-bot] ! Multi Contas / Mapeamento de Baus / Escolha Heroes / Correções (PR #647)


Ele funciona com um pc com mais de uma tela ?

—
Reply to this email directly, view it on GitHub<https://github.com/mpcabete/bombcrypto-bot/pull/647#issuecomment-1018589677>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AXFT5XY7WTRTP4JZC443JWDUXFYUXANCNFSM5MBNL6LA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
You are receiving this because you modified the open/close state.Message ID: ***@***.***>

ipucoinadmin commented:
> Conforme já mencionado acima o computador tem de ficar exclusivo pro bot durante o jogo multiconta. Utilizando 2 telas e usando o computador ao mesmo tempo o bot deve apresentar exceções reiniciado a rotina do bot. Obter o Outlook para Android<https://aka.ms/AAb9ysg>
> […](#)
> ________________________________ From: ipucoinadmin ***@***.***> Sent: Friday, January 21, 2022 12:03:39 PM To: mpcabete/bombcrypto-bot ***@***.***> Cc: bnudaniel22 ***@***.***>; State change ***@***.***> Subject: Re: [mpcabete/bombcrypto-bot] ! Multi Contas / Mapeamento de Baus / Escolha Heroes / Correções (PR #647) Ele funciona com um pc com mais de uma tela ? — Reply to this email directly, view it on GitHub<[#647 (comment)](https://github.com/mpcabete/bombcrypto-bot/pull/647#issuecomment-1018589677)>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AXFT5XY7WTRTP4JZC443JWDUXFYUXANCNFSM5MBNL6LA>. Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. You are receiving this because you modified the open/close state.Message ID: ***@***.***>

Eu deixei o computador exclusivo, deixei alguns minutos e nada.
Ele da alt+tab, mas a movimentação do mouse não funciona corretamente.
Em um outro tópico, explica melhor como fazer funcionar para vários monitores, vou procurar e tentar adequar.

kendikaku commented:
> > Conforme já mencionado acima o computador tem de ficar exclusivo pro bot durante o jogo multiconta. Utilizando 2 telas e usando o computador ao mesmo tempo o bot deve apresentar exceções reiniciado a rotina do bot. Obter o Outlook para Androidhttps://aka.ms/AAb9ysg
> > […](#)
> > ________________________________ From: ipucoinadmin _**@**_._**> Sent: Friday, January 21, 2022 12:03:39 PM To: mpcabete/bombcrypto-bot _**@**_.**_> Cc: bnudaniel22 _**@**_._**>; State change _**@**_.**_> Subject: Re: [mpcabete/bombcrypto-bot] ! Multi Contas / Mapeamento de Baus / Escolha Heroes / Correções (PR #647) Ele funciona com um pc com mais de uma tela ? — Reply to this email directly, view it on GitHub<[#647 (comment)](https://github.com/mpcabete/bombcrypto-bot/pull/647#issuecomment-1018589677)>, or unsubscribehttps://github.com/notifications/unsubscribe-auth/AXFT5XY7WTRTP4JZC443JWDUXFYUXANCNFSM5MBNL6LA. Triage notifications on the go with GitHub Mobile for iOShttps://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675 or Androidhttps://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub. You are receiving this because you modified the open/close state.Message ID: _**@**_.***>
> 
> Eu deixei o computador exclusivo, deixei alguns minutos e nada. Ele da alt+tab, mas a movimentação do mouse não funciona corretamente. Em um outro tópico, explica melhor como fazer funcionar para vários monitores, vou procurar e tentar adequar.

Usa esse bot

https://github.com/MatheusHAS/bombcrypto-bot

Estou usando pra 2 contas e está perfeito.

Suedera commented:
Boa noite, amigo.

Consigo desativar as funçoes de multi account? se sim, como? 
gostei bastante do código e gostaria de usar apenas em uma conta, é possível?

Juninhox13 commented:
opa mano, o bot é perfeito, só ta clicando no espaço entre os herois depois q ele scrolla, ja procurei nos outros topicos e fiz tudo q pediram la mas nao funcionou, pediram até pra alterar um 'offset' na linha 270 no index, porem a linha 270 la é uma linha em branco. mexi em resoluçao, nos prints do bot e nada, ele continua scrollando e clicando fora da caixinha verde do go work

bnudaniel22 commented:
> opa mano, o bot é perfeito, só ta clicando no espaço entre os herois depois q ele scrolla, ja procurei nos outros topicos e fiz tudo q pediram la mas nao funcionou, pediram até pra alterar um 'offset' na linha 270 no index, porem a linha 270 la é uma linha em branco. mexi em resoluçao, nos prints do bot e nada, ele continua scrollando e clicando fora da caixinha verde do go work

Olha estou utilizando o parâmetro mandar todos trabalhar (time 75 min),  a escolha de baús não está obtendo melhora no Farm, acaba perdendo muito tempo na leitura de baús  e escolha de heróis e não de jogo rodando
